Mathlib/Data/Polynomial/Degree/Lemmas.lean

```diff
@@ -289,6 +289,27 @@ theorem coe_lt_degree {p : R[X]} {n : ℕ} : (n : WithBot ℕ) < degree p ↔ n
   simp [degree_eq_natDegree h, Nat.cast_lt]
 #align polynomial.coe_lt_degree Polynomial.coe_lt_degree
 
+@[simp]
+theorem Polynomial.degree_map_eq_iff {R : Type u} {S : Type v}
+  [Semiring R] [Semiring S] {f : R →+* S} (p : Polynomial R) :
+    Polynomial.degree (Polynomial.map f p) = Polynomial.degree p
+      ↔
+    f (p.leadingCoeff) ≠ 0 := by
```

Comment:
Isn't the LHS true when `p = 0` but RHS false?

Reply:
Yep, seems like it, someone should fix that if they adopt this